api/api_full.go

```diff
@@ -236,7 +237,7 @@ type FullNode interface {
 	// ClientMinerQueryOffer returns a QueryOffer for the specific miner and file.
 	ClientMinerQueryOffer(ctx context.Context, miner address.Address, root cid.Cid, piece *cid.Cid) (QueryOffer, error)
 	// ClientRetrieve initiates the retrieval of a file, as specified in the order.
-	ClientRetrieve(ctx context.Context, order RetrievalOrder, ref *FileRef) error
+	ClientRetrieve(ctx context.Context, order RetrievalOrder, ref *FileRef) (<-chan marketevents.RetrievalEvent, error)
```

Comment:
It may be worth having a version of this call without events, for people using jsonrpc libs without channel support / over http (`ClientRetrieve` / `ClientRetrieveWithEvents`)

Reply:
I will put this back in in a followup PR